### PR TITLE
Mon 4623 remote config

### PR DESCRIPTION
**Pull request by loiclau**
## Description

When I did define a Poller that uses ZMQ communication type, an icon appears in the “Actions” column of the monitoring instances listing:
When I click on this icon, a popin shows me the actions to perform to export the configuration.

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [ ] 19.10.x
- [x] 20.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [x] I followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/include/Administration/parameters/gorgone/gorgone.php [hunk 1]
```diff
@@ -57,6 +57,27 @@
...
+// API
+$form->addElement('text', 'gorgone_api_address', _("Address of Gorgone API"), $attrsText);
+$form->addElement('text', 'gorgone_api_port', _("Port of Gorgone API"), $attrsText2);
+$form->addRule('gorgone_api_port', _('Must be a number'), 'numeric');
+$form->addElement('text', 'gorgone_api_username', _("Username of Gorgone API"), $attrsText);
+$form->addElement('password', 'gorgone_api_password', _("Password of Gorgone API"), $attrsText);
+$form->addElement(
+    'checkbox',
+    'gorgone_api_ssl',
+    _("Use SSH for Gorgone API"),
```
Comment (cgagnaire) [inline www/include/Administration/parameters/gorgone/gorgone.php]: SSL not SSH

File: www/install/sql/centreon/Update-DB-20.04.0-beta.1.sql [hunk 2]
```diff
@@ -28,4 +28,8 @@ UPDATE options SET `key` = 'gorgone_cmd_timeout' WHERE `key` = 'centcore_cmd_tim
 UPDATE topology SET topology_url_opt = '&o=gorgone', topology_name = 'Gorgone' WHERE topology_page = 50117;
 UPDATE options SET `key` = 'debug_gorgone' WHERE `key` = 'debug_centcore';
 DELETE FROM `options` WHERE `key` = 'enable_perfdata_sync';
-DELETE FROM `options` WHERE `key` = 'enable_logs_sync';
\ No newline at end of file
+DELETE FROM `options` WHERE `key` = 'enable_logs_sync';
+-- gorgone API default
+INSERT INTO `options` (`key`, `value`) VALUES ('gorgone_api_address', '127.0.0.1');
+INSERT INTO `options` (`key`, `value`) VALUES ('gorgone_api_port', '8085');
+INSERT INTO `options` (`key`, `value`) VALUES ('gorgone_api_ssl', '0');
```
Comment (cgagnaire) [inline www/install/sql/centreon/Update-DB-20.04.0-beta.1.sql]: New `gorgone_api_allow_self_signed` key is missing right ?

File: www/include/Administration/parameters/gorgone/gorgone.php (hunk 1 above)
Comment (loiclau) [inline www/include/Administration/parameters/gorgone/gorgone.php]:
```suggestion
    _("Use SSL for Gorgone API"),
```

File: www/install/sql/centreon/Update-DB-20.04.0-beta.1.sql (hunk 2 above)
Comment (loiclau) [inline www/install/sql/centreon/Update-DB-20.04.0-beta.1.sql]:
```suggestion
INSERT INTO `options` (`key`, `value`) VALUES ('gorgone_api_ssl', '0');
INSERT INTO `options` (`key`, `value`) VALUES ('gorgone_api_allow_self_signed', '1');
```

File: www/include/Administration/parameters/DB-Func.php [hunk 3]
```diff
@@ -295,6 +295,37 @@ function updateGorgoneConfigData($db, $form, $centreon)
...
+    );
+    updateOption(
+        $db,
+        'gorgone_api_ssl',
+        $ret['gorgone_api_ssl'] ?? '0'
+    );
+    updateOption(
+        $db,
+        'gorgone_api_allow_self_signed',
+        $ret['gorgone_api_allow_self_signed'] ?? '0'
```
Comment (loiclau) [inline www/include/Administration/parameters/DB-Func.php]:
```suggestion
        $ret['gorgone_api_allow_self_signed'] ?? '1'
```

File: src/Kernel.php [hunk 4]
```diff
@@ -48,7 +48,7 @@ class Kernel extends BaseKernel
     public static function createForWeb(): Kernel
     {
         if (static::$instance === null) {
-            include_once '../config/bootstrap.php';
+            include_once _CENTREON_PATH_ . '/config/bootstrap.php';
```
Comment (kduret) [inline src/Kernel.php]: Should be better to use __DIR__ constant

File: www/include/configuration/configServers/listServers.php [hunk 5]
```diff
@@ -257,31 +257,37 @@
...
+        'RowMenu_badge' => $config['ns_activate'] ? 'service_ok' : 'service_critical',
+        'RowMenu_statusVal' => $config['ns_activate'],
+        'RowMenu_cfg_id' => ($cfg_id == -1) ? '' : $cfg_id['nagios_id'],
+        'RowMenu_options' => $moptions
     ];
 }
 $tpl->assign("elemArr", $elemArr);
 
+
+//$form->addElement('button', 'desc_arg', _("Describe arguments"), array("onClick" => "goPopup();"));
```
Comment (kduret) [inline www/include/configuration/configServers/listServers.php]: Remove comments

File: www/include/configuration/configServers/popup/popup.php [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+        $dataError .= '
+      - error : Poller ' . $serverId . ' : ' . $thummprintReponse['message'];
+    }
+}
+
+if (!empty($dataError)) {
+    $config = $dataError;
+} elseif (in_array($server['ns_ip_address'], $remotesServerIPs)) {
+    //config for remote
+    $config = 'name: gorgoned-' . $server['name'] . '
```
Comment (kduret) [inline www/include/configuration/configServers/popup/popup.php]: Could you extract this in a yaml template file ?

File: www/include/configuration/configServers/popup/popup.php [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+
+    - name: engine
+      package: gorgone::modules::centreon::engine::hooks
+      enable: true
+      command_file: "' . $server['command_file'] . '"
+
+';
+} else {
+    //config for poller
+    $config = 'name:  gorgoned-' . $server['name'] . '
```
Comment (kduret) [inline www/include/configuration/configServers/popup/popup.php]: Could you extract this in a yaml template file ?

File: www/include/configuration/configServers/popup/popup.ihtml [hunk 8]
```diff
@@ -0,0 +1,35 @@
+<div>
+	<h3 style="text-align: center;"> Gorgone configuration file :</h3>
+    <br>
+    {if $gorgoneError == '0'}
+	<p style="text-align: center;">{$gorgoneError}
+        <strong>
+            Replace the '/etc/centreon/config.d/20-gorgoned.yaml' configuration file on your server {$serverIp}
```
Comment (sc979) [inline www/include/configuration/configServers/popup/popup.ihtml]:
```suggestion
            Replace the '/etc/centreon-gorgone/config.d/40-gorgoned.yaml' configuration file on your server {$serverIp}
```

File: www/include/configuration/configServers/popup/poller.yaml [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+    authorized_clients: __THUMBPRINT__
+    privkey: "/var/lib/centreon-gorgone/.keys/rsakey.priv.pem"
+    pubkey: "/var/lib/centreon-gorgone/.keys/rsakey.pub.pem"
+  modules:
+    - name: action
+      package: gorgone::modules::core::action::hooks
+      enable: true
+
+    - name: engine
+      package: gotrgone::modules::centreon::engine::hooks
```
Comment (sc979) [inline www/include/configuration/configServers/popup/poller.yaml]:
```suggestion
      package: gorgone::modules::centreon::engine::hooks
```

File: www/include/configuration/configServers/popup/remote.yaml [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+      enable: true
+
+    - name: proxy
+      package: gorgone::modules::core::proxy::hooks
+      enable: true
+
+    - name: legacycmd
+      package: gorgone::modules::centreon::legacycmd::hooks
+      enable: true
+      cmd_file: "/var/lib/centreon/centcore.cmd"
```
Comment (sc979) [inline www/include/configuration/configServers/popup/remote.yaml]: is it possible to choose another path (eg : source installation script on a poller)

**Pull request merged**